File: src/main/kotlin/solutionsJava/Repetitions2.java
import java.util.Scanner;

public class Repetitions2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String str = scanner.nextLine().trim();

        System.out.println(longestSubstringWithoutRepeatingChar(str));
    }

    private static int longestSubstringWithoutRepeatingChar(String str) {
        int left = 0;
        int right = 0;
        int res = 0;

        while (right < str.length()) {
            if (str.charAt(right) != str.charAt(left)) {
                res = Math.max(res, right - left);
                left = right;
            }
            right++;
        }
        res = Math.max(res, right - left);

        return res;
    }
}

File: src/main/kotlin/solutions/Repetitions2.kt
package uz.safix.solutions

import java.util.Scanner

fun main() {
    val scanner = Scanner(System.`in`)
    val str = scanner.nextLine().trim()

    println(longestSubstringWithoutRepeatingChar(str))
}

private fun longestSubstringWithoutRepeatingChar(str: String): Int {
    var left = 0
    var right = 0
    var res = 0

    while (right < str.length) {
        if (str[right] != str[left]) {
            res = maxOf(res, right - left)
            left = right
        }
        right++
    }
    res = maxOf(res, right - left)

    return res
}